lib/connection.c

```diff
@@ -769,11 +796,24 @@ sf_bool STDCALL http_perform(CURL *curl,
 
     //TODO set error buffer
 
+    // Find request GUID in the supplied URL
+    char *request_guid_ptr = strstr(url, "request_guid=");
+    // Set pointer to the beginning of the UUID string if request GUID exists
+    if (request_guid_ptr) {
+        request_guid_ptr = request_guid_ptr + 13;
+    }
+
     do {
         // Reset buffer since this may not be our first rodeo
         SF_FREE(buffer.buffer);
         buffer.size = 0;
 
+        // Generate new request guid, if request guid exists in url
+        if (request_guid_ptr && uuid4_generate_non_terminated(request_guid_ptr)) {
```

Comment:
This would be more clear that uuid4 generation failed.
```
if (request_guid_ptr && uuid4_generate_non_terminated(request_guid_ptr) != UUID4_ESUCCESS)
```